core/pom.xml

```diff
@@ -180,6 +206,23 @@
       <version>1.6.0</version>
     </dependency>
 
+    <dependency>
+      <groupId>org.graalvm.nativeimage</groupId>
+      <artifactId>svm</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.graalvm.sdk</groupId>
+      <artifactId>graal-sdk</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <!-- GAX provides native-image metadata for core Google libraries. GAX's
+       opencensus-api, protobuf-java, -->
+      <groupId>com.google.api</groupId>
+      <artifactId>gax</artifactId>
+    </dependency>
```

Comment:
Same here.

Reply:
This library needs to be in the class path when library users run GraalVM native image compilation. If we make this as a profile, the users don't get the dependency.

Reply:
Added a source-code comment to explain that.

Reply:
I have little no experience with Graal, so let me ask some maybe dumb questions:
1. Why is this better off as `provided` as opposed to `runtime`? 
2. Is this how all of our libraries are adding support for Graal? What about users that don't use Graal - isn't this just putting unnecessary dependencies in the path?

Reply:
> Why is this better off as provided as opposed to runtime?

It's because the classes `svm` and `graal-sdk` are available in GraalVM native image compiler.

The "runtime" scope requires the application to get the dependency. For example, `com.github.jnr:jffi` is a runtime-scope dependency in jdbc/postgres module. When Maven creates "jar-with-driver-and-dependencies", it includes the content of the runtime dependency:

![image](https://user-images.githubusercontent.com/28604/165774370-b5498266-c016-4b63-ab83-47e77db2fb09.png)

The "provided" scope does not require the application to get the dependency. When an application runs, the enclosing environment (such as Tomcat, GraalVM native image compiler, etc.) provides the classes. Therefore provided-scope dependencies are not included in the jar-with-driver-and-dependencies:

```
suztomo@suztomo2:~/cloud-sql-jdbc-socket-factory/jdbc/postgres$ jar tf target/postgres-socket-factory-1.5.1-SNAPSHOT-jar-with-driver-and-dependencies.jar |grep AutomaticFeature
suztomo@suztomo2:~/cloud-sql-jdbc-socket-factory/jdbc/postgres$ jar tf target/postgres-socket-factory-1.5.1-SNAPSHOT-jar-with-driver-and-dependencies.jar |grep com/oracle
suztomo@suztomo2:~/cloud-sql-jdbc-socket-factory/jdbc/postgres$ jar tf target/postgres-socket-factory-1.5.1-SNAPSHOT-jar-with-driver-and-dependencies.jar |grep com/graalvm
suztomo@suztomo2:~/cloud-sql-jdbc-socket-factory/jdbc/postgres$ 
```

(AutomaticFeature, com.oracle, and com.graalvm are the class and packages inside `svm` and `graal-sdk`)


> Is this how all of our libraries are adding support for Graal? What about users that don't use Graal - isn't this just putting unnecessary dependencies in the path?

Yes, for those who don't use GraalVM native image, it's unnecessary dependency. I just added `<exclusions>` to reduce the amount of them. The additional overhead of gax-2.16.0.jar is 376 KB.

```
suztomo@suztomo2:~/cloud-sql-jdbc-socket-factory/jdbc/postgres$ ls -alt ~/.m2/repository/com/google/api/gax/2.16.0/gax-2.16.0.jar
-rw-r--r-- 1 suztomo primarygroup 376419 Apr 21 10:15 /usr/local/google/home/suztomo/.m2/repository/com/google/api/gax/2.16.0/gax-2.16.0.jar
suztomo@suztomo2:~/cloud-sql-jdbc-socket-factory/jdbc/postgres$ ls -alt target/postgres-socket-factory-1.5.1-SNAPSHOT-jar-with-driver-and-dependencies.jar
-rw-r--r-- 1 suztomo primarygroup 12056609 Apr 28 10:18 target/postgres-socket-factory-1.5.1-SNAPSHOT-jar-with-driver-and-dependencies.jar
```

Reply:
Ok, thanks for the detailed followup.